nucypher/blockchain/eth/deployers.py

```diff
@@ -183,21 +183,33 @@ class DispatcherDeployer(ContractDeployer):
     """
 
     contract_name = 'Dispatcher'
-    __upgradeable = False
+    _upgradeable = False
 
     DISPATCHER_SECRET_LENGTH = 32
 
-    def __init__(self, target_contract: Contract, secret_hash: bytes, *args, **kwargs):
+    def __init__(self, target_contract: Contract, bare: bool = False, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.target_contract = target_contract
-        self.secret_hash = secret_hash
+        if bare:
```

Comment:
Hmmm, not really sure what `bare` means in this context. Can you elaborate a bit here?

Reply:
Hehe - Yeah - I need an option to instantiate a deployer *without* contract agency, but if we are upgrading a contract, then *with* agency.

Reply:
Could you perhaps use `has_agency` or something to that effect? I feel like `bare` isn't that apparent.